embassy-nrf/src/spim.rs

```diff
@@ -234,7 +268,18 @@ impl<'d, T: Instance> Spim<'d, T> {
         Ok(())
     }
 
-    async fn async_inner(&mut self, rx: *mut [u8], tx: *const [u8]) -> Result<(), Error> {
+    fn blocking_inner(&mut self, rx: &mut [u8], tx: &[u8]) -> Result<(), Error> {
+        match self.blocking_inner_from_ram(rx, tx) {
+            Ok(_) => Ok(()),
+            Err(Error::DMABufferNotInDataMemory) => {
+                let tx_copied = tx.clone();
```

Comment:
`.clone()` here clones just the slice pointer, not the data. The data will still be in Flash.

You need to copy through a buffer in the stack. This is how nrf-hal does it for example: https://github.com/nrf-rs/nrf-hal/blob/master/nrf-hal-common/src/spim.rs#L94

Reply:
I was wondering about that whether it would do that. Thanks for the pointer (pun very much intended)!

Reply:
I was initially trying to do it that way but having a fixed-size buffer allocation of 255-1024 byte seems like a pretty hefty price to pay. Will the compiler optimise it away if it is not required (either by not calling the "regular" methods or only calling it with in-memory checks)?

Reply:
For blocking, if the buffer is in a fn that is only called when copying, it should only use the extra memory if it does get called... For async it'll probably use the memory no matter what, the size of a future is the max of all possible child futures...

Reply:
It is what it is :(

Reply:
Either way, I've changed it to use a buffer based on the `chip::FORCE_COPY_BUFFER_SIZE` constant that was already present.